tests/bud.bats

```diff
@@ -2811,3 +2816,301 @@ _EOF
+from alpine
+run printf "cpuset-cpus " && cat /sys/fs/cgroup/\$(cat /proc/self/cgroup | awk -F : '{print \$NF}')/cpuset.cpus
+_EOF
+  else
+    cat > $mytmpdir/Containerfile << _EOF
+from alpine
+run printf "cpuset-cpus " && cat /sys/fs/cgroup/cpuset/cpuset.cpus
+_EOF
+  fi
+
+  run_buildah bud --cpuset-cpus=0 -t testcpuset \
+                  --signature-policy ${TESTSDIR}/policy.json --file ${mytmpdir} .
+  expect_output --substring "cpuset-cpus 0"
+}
+
+@test "bud with --cpuset-mems" {
+  skip_if_chroot
+  skip_if_rootless
+  skip_if_no_runtime
+  skip_if_cgroupsv2
+
+  _prefetch alpine
+
+  mytmpdir=${TESTDIR}/my-dir
+  mkdir -p ${mytmpdir}
+  cat > $mytmpdir/Containerfile << _EOF
+from alpine
+run printf "cpuset-mems " && cat /sys/fs/cgroup/cpuset/cpuset.mems
+_EOF
+
+  run_buildah bud --cpuset-mems=0 -t testcpuset \
+                  --signature-policy ${TESTSDIR}/policy.json --file ${mytmpdir} .
+  expect_output --substring "cpuset-mems 0"
+}
+
+@test "bud with --isolation" {
+  skip_if_no_runtime
+  test -z "${BUILDAH_ISOLATION}" || skip "BUILDAH_ISOLATION=${BUILDAH_ISOLATION} overrides --isolation"
+
+  _prefetch alpine
+
+  mytmpdir=${TESTDIR}/my-dir
+  mkdir -p ${mytmpdir}
+  cat > $mytmpdir/Containerfile << _EOF
+from alpine
+run readlink /proc/self/ns/pid
+_EOF
+
+  run readlink /proc/self/ns/pid
+  host_pidns=$output
+  run_buildah bud --isolation chroot -t testisolation --pid private \
+                  --signature-policy ${TESTSDIR}/policy.json --file ${mytmpdir} .
+  # chroot isolation doesn't make a new PID namespace.
+  expect_output --substring $(printf %q ${host_pidns})
+}
+
+@test "bud with --pull-always" {
+  _prefetch docker.io/library/alpine
+  run_buildah bud --pull-always --signature-policy ${TESTSDIR}/policy.json -t testpull ${TESTSDIR}/bud/containerfile
+  expect_output --substring "Getting image source signatures"
+}
+
+@test "bud with --memory and --memory-swap" {
+  skip_if_chroot
+  skip_if_no_runtime
+  skip_if_rootless
+
+  _prefetch alpine
+
+  mytmpdir=${TESTDIR}/my-dir
+  mkdir -p ${mytmpdir}
+
+  local expect_swap=
+  if is_cgroupsv2; then
+    cat > $mytmpdir/Containerfile << _EOF
+from alpine
+run printf "memory-max=" && cat /sys/fs/cgroup/\$(cat /proc/self/cgroup | awk -F : '{print \$NF}')/memory.max
+run printf "memory-swap-result=" && cat /sys/fs/cgroup/\$(cat /proc/self/cgroup | awk -F : '{print \$NF}')/memory.swap.max
+_EOF
+    expect_swap=31457280
+  else
+    cat > $mytmpdir/Containerfile << _EOF
+from alpine
+run printf "memory-max=" && cat /sys/fs/cgroup/memory/memory.limit_in_bytes
+run printf "memory-swap-result=" && cat /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes
+_EOF
+    expect_swap=73400320
+  fi
+
+  run_buildah bud --memory=40m --memory-swap=70m -t testmemory \
+                  --signature-policy ${TESTSDIR}/policy.json --file ${mytmpdir} .
+  expect_output --substring "memory-max=41943040"
+  expect_output --substring "memory-swap-result=${expect_sw}"
```

Comment:
typo: the variable is actually `expect_swap`. This is passing because `${expect_sw}` is empty. See why I try to avoid `--substring` whenever possible? Please rewrite this to use `--from="${lines[...]}"` and exact matches.

Reply:
Thank you for catching it! I agree that I should avoid using '--substring' as much as possible.